### PR TITLE
feat!: add IPv6 support

### PR DESCRIPTION
**Pull request by Lynty**
https://github.com/terraform-google-modules/terraform-google-network/issues/338

Added to @anthr76's work: https://github.com/terraform-google-modules/terraform-google-network/pull/397 (Signed-off-by: Anthony Rabbito [hello@anthonyrabbito.com](mailto:hello@anthonyrabbito.com))

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/README.md /tmp/tmp.aJsief0iLj/generate_docs/workspace/README.md
105a106
> | ipv6\_access\_type | The access type of IPv6 address this subnet holds. It's immutable and can only be specified during creation or the first time the subnet is updated into IPV4\_IPV6 dual stack. If the ipv6\_type is EXTERNAL then this subnet cannot enable direct path. | `string` | n/a | yes |
112a114
> | stack\_type | The stack type for this subnet to identify whether the IPv6 feature is enabled or not. If not specified IPV4\_ONLY will be used. | `string` | n/a | yes |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_tflint
 Failed TFlint check. More details below. 
 ```bash 
Checking for tflint
Working in . ...
2 issue(s) found:
Warning: variable "stack_type" is declared but not used (terraform_unused_declarations)
  on variables.tf line 87:
  87: variable "stack_type" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_unused_declarations.md
Warning: variable "ipv6_access_type" is declared but not used (terraform_unused_declarations)
  on variables.tf line 92:
  92: variable "ipv6_access_type" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_unused_declarations.md
tflint failed . 
Working in ./codelabs/simple ...
tflint passed ./codelabs/simple 
Working in ./examples/basic_auto_mode ...
tflint passed ./examples/basic_auto_mode 
Working in ./examples/basic_custom_mode ...
tflint passed ./examples/basic_custom_mode 
Working in ./examples/basic_firewall_rule ...
tflint passed ./examples/basic_firewall_rule 
Working in ./examples/basic_secondary_ranges ...
tflint passed ./examples/basic_secondary_ranges 
Working in ./examples/basic_shared_vpc ...
tflint passed ./examples/basic_shared_vpc 
Working in ./examples/basic_vpc_peering ...
tflint passed ./examples/basic_vpc_peering 
Working in ./examples/delete_default_gateway_routes ...
tflint passed ./examples/delete_default_gateway_routes 
Working in ./examples/firewall_logging ...
tflint passed ./examples/firewall_logging 
Working in ./examples/ilb_routing ...
tflint passed ./examples/ilb_routing 
Working in ./examples/multi_vpc ...
tflint passed ./examples/multi_vpc 
Working in ./examples/network_service_tiers ...
tflint passed ./examples/network_service_tiers 
Working in ./examples/packet_mirroring ...
tflint passed ./examples/packet_mirroring 
Working in ./examples/private_service_connect ...
tflint passed ./examples/private_service_connect 
Working in ./examples/private_service_connect_google_apis ...
tflint passed ./examples/private_service_connect_google_apis 
Working in ./examples/routes ...
tflint passed ./examples/routes 
Working in ./examples/secondary_ranges ...
tflint passed ./examples/secondary_ranges 
Working in ./examples/simple_ipv6_project ...
tflint passed ./examples/simple_ipv6_project 
Working in ./examples/simple_project ...
tflint passed ./examples/simple_project 
Working in ./examples/simple_project_with_regional_network ...
tflint passed ./examples/simple_project_with_regional_network 
Working in ./examples/submodule_firewall ...
tflint passed ./examples/submodule_firewall 
Working in ./examples/submodule_network_peering ...
tflint passed ./examples/submodule_network_peering 
Working in ./examples/submodule_svpc_access ...
tflint passed ./examples/submodule_svpc_access 
Working in ./examples/submodule_vpc_serverless_connector ...
tflint passed ./examples/submodule_vpc_serverless_connector 
Working in ./modules/fabric-net-firewall ...
tflint passed ./modules/fabric-net-firewall 
Working in ./modules/fabric-net-svpc-access ...
tflint passed ./modules/fabric-net-svpc-access 
Working in ./modules/firewall-rules ...
tflint passed ./modules/firewall-rules 
Working in ./modules/network-peering ...
tflint passed ./modules/network-peering 
Working in ./modules/private-service-connect ...
tflint passed ./modules/private-service-connect 
Working in ./modules/routes ...
tflint passed ./modules/routes 
Working in ./modules/routes-beta ...
tflint passed ./modules/routes-beta 
Working in ./modules/subnets ...
tflint passed ./modules/subnets 
Working in ./modules/subnets-beta ...
tflint passed ./modules/subnets-beta 
Working in ./modules/vpc ...
tflint passed ./modules/vpc 
Working in ./modules/vpc-serverless-connector-beta ...
tflint passed ./modules/vpc-serverless-connector-beta 
Skipping ./test/fixtures/delete_default_gateway_routes
Skipping ./test/fixtures/ilb_routing
Skipping ./test/fixtures/multi_vpc
Skipping ./test/fixtures/secondary_ranges
Skipping ./test/fixtures/simple_project
Skipping ./test/fixtures/simple_project_with_regional_network
Skipping ./test/fixtures/submodule_firewall
Skipping ./test/setup
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate .
Success!
 The configuration is valid.
terraform_validate ./codelabs/simple
Success!
 The configuration is valid.
terraform_validate ./examples/basic_auto_mode
Success!
 The configuration is valid.
terraform_validate ./examples/basic_custom_mode
Success!
 The configuration is valid.
terraform_validate ./examples/basic_firewall_rule
Success!
 The configuration is valid.
terraform_validate ./examples/basic_secondary_ranges
Success!
 The configuration is valid.
terraform_validate ./examples/basic_shared_vpc
Success!
 The configuration is valid.
terraform_validate ./examples/basic_vpc_peering
Success!
 The configuration is valid.
terraform_validate ./examples/delete_default_gateway_routes
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 21, in module "test-vpc-module":
│
 
  21: module "test-vpc-module" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 21, in module "test-vpc-module":
│
 
  21: module "test-vpc-module" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
terraform_validate ./examples/firewall_logging
Success!
 The configuration is valid.
terraform_validate ./examples/ilb_routing
Success!
 The configuration is valid.
terraform_validate ./examples/multi_vpc
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 104, in module "test-vpc-module-02":
│
 
 104: module "test-vpc-module-02" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 104, in module "test-vpc-module-02":
│
 
 104: module "test-vpc-module-02" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
terraform_validate ./examples/network_service_tiers
Success!
 The configuration is valid.
terraform_validate ./examples/packet_mirroring
Success!
 The configuration is valid.
terraform_validate ./examples/private_service_connect
Success!
 The configuration is valid.
terraform_validate ./examples/private_service_connect_google_apis
Success!
 The configuration is valid.
terraform_validate ./examples/routes
Success!
 The configuration is valid.
terraform_validate ./examples/secondary_ranges
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 24, in module "vpc-secondary-ranges":
│
 
  24: module "vpc-secondary-ranges" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 24, in module "vpc-secondary-ranges":
│
 
  24: module "vpc-secondary-ranges" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
terraform_validate ./examples/simple_ipv6_project
Success!
 The configuration is valid.
terraform_validate ./examples/simple_project
Success!
 The configuration is valid.
terraform_validate ./examples/simple_project_with_regional_network
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 22, in module "test-vpc-module":
│
 
  22: module "test-vpc-module" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 22, in module "test-vpc-module":
│
 
  22: module "test-vpc-module" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
terraform_validate ./examples/submodule_firewall
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 22, in module "test-vpc-module":
│
 
  22: module "test-vpc-module" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 22, in module "test-vpc-module":
│
 
  22: module "test-vpc-module" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
terraform_validate ./examples/submodule_network_peering
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 17, in module "local-network":
│
 
  17: module "local-network" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 17, in module "local-network":
│
 
  17: module "local-network" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
terraform_validate ./examples/submodule_svpc_access
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 24, in module "net-vpc-shared":
│
 
  24: module "net-vpc-shared" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on main.tf line 24, in module "net-vpc-shared":
│
 
  24: module "net-vpc-shared" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
terraform_validate ./examples/submodule_vpc_serverless_connector
Success!
 The configuration is valid.
terraform_validate ./modules/fabric-net-firewall
Success!
 The configuration is valid.
terraform_validate ./modules/fabric-net-svpc-access
Success!
 The configuration is valid.
terraform_validate ./modules/firewall-rules
Success!
 The configuration is valid.
terraform_validate ./modules/network-peering
Success!
 The configuration is valid.
terraform_validate ./modules/private-service-connect
Success!
 The configuration is valid.
terraform_validate ./modules/routes
Success!
 The configuration is valid.
terraform_validate ./modules/routes-beta
Success!
 The configuration is valid.
terraform_validate ./modules/subnets
Success!
 The configuration is valid.
terraform_validate ./modules/subnets-beta
Success!
 The configuration is valid.
terraform_validate ./modules/vpc
Success!
 The configuration is valid.
terraform_validate ./modules/vpc-serverless-connector-beta
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/delete_default_gateway_routes
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/delete_default_gateway_routes/main.tf line 21, in module "test-vpc-module":
│
 
  21: module "test-vpc-module" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/delete_default_gateway_routes/main.tf line 21, in module "test-vpc-module":
│
 
  21: module "test-vpc-module" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
terraform_validate ./test/fixtures/ilb_routing
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/multi_vpc
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/multi_vpc/main.tf line 52, in module "test-vpc-module-01":
│
 
  52: module "test-vpc-module-01" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/multi_vpc/main.tf line 52, in module "test-vpc-module-01":
│
 
  52: module "test-vpc-module-01" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
terraform_validate ./test/fixtures/secondary_ranges
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/secondary_ranges/main.tf line 24, in module "vpc-secondary-ranges":
│
 
  24: module "vpc-secondary-ranges" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/secondary_ranges/main.tf line 24, in module "vpc-secondary-ranges":
│
 
  24: module "vpc-secondary-ranges" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
terraform_validate ./test/fixtures/simple_project
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_project_with_regional_network
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/simple_project_with_regional_network/main.tf line 22, in module "test-vpc-module":
│
 
  22: module "test-vpc-module" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/simple_project_with_regional_network/main.tf line 22, in module "test-vpc-module":
│
 
  22: module "test-vpc-module" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
terraform_validate ./test/fixtures/submodule_firewall
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/submodule_firewall/main.tf line 22, in module "test-vpc-module":
│
 
  22: module "test-vpc-module" 
{
│
 
│
 
The argument "ipv6_access_type" is required, but no definition was found.
╵
╷
│
 
Error: 
Missing required argument
│
 
│
 
  on ../../../examples/submodule_firewall/main.tf line 22, in module "test-vpc-module":
│
 
  22: module "test-vpc-module" 
{
│
 
│
 
The argument "stack_type" is required, but no definition was found.
╵
terraform_validate ./test/setup
Success!
 The configuration is valid.
 ```

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_tflint
 Failed TFlint check. More details below. 
 ```bash 
Checking for tflint
Working in . ...
2 issue(s) found:
Warning: variable "stack_type" is declared but not used (terraform_unused_declarations)
  on variables.tf line 87:
  87: variable "stack_type" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_unused_declarations.md
Warning: variable "ipv6_access_type" is declared but not used (terraform_unused_declarations)
  on variables.tf line 93:
  93: variable "ipv6_access_type" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_unused_declarations.md
tflint failed . 
Working in ./codelabs/simple ...
tflint passed ./codelabs/simple 
Working in ./examples/basic_auto_mode ...
tflint passed ./examples/basic_auto_mode 
Working in ./examples/basic_custom_mode ...
tflint passed ./examples/basic_custom_mode 
Working in ./examples/basic_firewall_rule ...
tflint passed ./examples/basic_firewall_rule 
Working in ./examples/basic_secondary_ranges ...
tflint passed ./examples/basic_secondary_ranges 
Working in ./examples/basic_shared_vpc ...
tflint passed ./examples/basic_shared_vpc 
Working in ./examples/basic_vpc_peering ...
tflint passed ./examples/basic_vpc_peering 
Working in ./examples/delete_default_gateway_routes ...
tflint passed ./examples/delete_default_gateway_routes 
Working in ./examples/firewall_logging ...
tflint passed ./examples/firewall_logging 
Working in ./examples/ilb_routing ...
tflint passed ./examples/ilb_routing 
Working in ./examples/multi_vpc ...
tflint passed ./examples/multi_vpc 
Working in ./examples/network_service_tiers ...
tflint passed ./examples/network_service_tiers 
Working in ./examples/packet_mirroring ...
tflint passed ./examples/packet_mirroring 
Working in ./examples/private_service_connect ...
tflint passed ./examples/private_service_connect 
Working in ./examples/private_service_connect_google_apis ...
tflint passed ./examples/private_service_connect_google_apis 
Working in ./examples/routes ...
tflint passed ./examples/routes 
Working in ./examples/secondary_ranges ...
tflint passed ./examples/secondary_ranges 
Working in ./examples/simple_ipv6_project ...
tflint passed ./examples/simple_ipv6_project 
Working in ./examples/simple_project ...
tflint passed ./examples/simple_project 
Working in ./examples/simple_project_with_regional_network ...
tflint passed ./examples/simple_project_with_regional_network 
Working in ./examples/submodule_firewall ...
tflint passed ./examples/submodule_firewall 
Working in ./examples/submodule_network_peering ...
tflint passed ./examples/submodule_network_peering 
Working in ./examples/submodule_svpc_access ...
tflint passed ./examples/submodule_svpc_access 
Working in ./examples/submodule_vpc_serverless_connector ...
tflint passed ./examples/submodule_vpc_serverless_connector 
Working in ./modules/fabric-net-firewall ...
tflint passed ./modules/fabric-net-firewall 
Working in ./modules/fabric-net-svpc-access ...
tflint passed ./modules/fabric-net-svpc-access 
Working in ./modules/firewall-rules ...
tflint passed ./modules/firewall-rules 
Working in ./modules/network-peering ...
tflint passed ./modules/network-peering 
Working in ./modules/private-service-connect ...
tflint passed ./modules/private-service-connect 
Working in ./modules/routes ...
tflint passed ./modules/routes 
Working in ./modules/routes-beta ...
tflint passed ./modules/routes-beta 
Working in ./modules/subnets ...
tflint passed ./modules/subnets 
Working in ./modules/subnets-beta ...
tflint passed ./modules/subnets-beta 
Working in ./modules/vpc ...
tflint passed ./modules/vpc 
Working in ./modules/vpc-serverless-connector-beta ...
tflint passed ./modules/vpc-serverless-connector-beta 
Skipping ./test/fixtures/delete_default_gateway_routes
Skipping ./test/fixtures/ilb_routing
Skipping ./test/fixtures/multi_vpc
Skipping ./test/fixtures/secondary_ranges
Skipping ./test/fixtures/simple_project
Skipping ./test/fixtures/simple_project_with_regional_network
Skipping ./test/fixtures/submodule_firewall
Skipping ./test/setup
 ```

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/README.md /tmp/tmp.koznGSeOvH/generate_docs/workspace/README.md
106d105
< | ipv6\_access\_type | The access type of IPv6 address this subnet holds. It's immutable and can only be specified during creation or the first time the subnet is updated into IPV4\_IPV6 dual stack. If the ipv6\_type is EXTERNAL then this subnet cannot enable direct path. | `string` | `"INTERNAL"` | no |
114d112
< | stack\_type | The stack type for this subnet to identify whether the IPv6 feature is enabled or not. | `string` | `"IPV4_ONLY"` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: examples/simple_ipv6_project/variables.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright 2019 Google LLC
```
Comment (apeabody) [inline examples/simple_ipv6_project/variables.tf:2]:
```suggestion
 * Copyright 2023 Google LLC
```

File: examples/simple_ipv6_project/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2019 Google LLC
```
Comment (apeabody) [inline examples/simple_ipv6_project/outputs.tf:2]:
```suggestion
 * Copyright 2023 Google LLC
```

File: examples/simple_ipv6_project/main.tf [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2019 Google LLC
```
Comment (apeabody) [inline examples/simple_ipv6_project/main.tf:2]:
```suggestion
 * Copyright 2023 Google LLC
```

File: examples/simple_ipv6_project/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Simple Project
```
Comment (apeabody) [inline examples/simple_ipv6_project/README.md:1]:
```suggestion
# Simple IPv6 Project
```

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @Lynty!  I've added a few comments.

@bharathkkb - As this new capability is exposed only in a new example, I'm assuming INT test coverage will be required?

File: examples/simple_ipv6_project/versions.tf [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+terraform {
+  required_version = ">=0.12.6"
+
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 4.25.0"
```
[the remark this answers is not in the document]
Comment (Lynty) [reply examples/simple_ipv6_project/versions.tf:23]: External IPv6 support was first introduced in [v3.88.0](https://github.com/hashicorp/terraform-provider-google/releases/tag/v3.88.0) and internal IPv6 support in [v4.25.0](https://github.com/hashicorp/terraform-provider-google/releases/tag/v4.25.0).

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: main.tf [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2019 Google LLC
+ * Copyright 2023 Google LLC
```
Comment (apeabody) [inline main.tf:2]: We can skip as this is an existing file.

```suggestion
 * Copyright 2019 Google LLC
```

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: main.tf (hunk 6 above)
Comment (Lynty) [reply main.tf:2]: whoops!

File: examples/simple_ipv6_project/versions.tf (hunk 5 above)
[the remark this answers is not in the document]
Comment (apeabody) [reply examples/simple_ipv6_project/versions.tf:23]: Thanks @Lynty! - In that case we should probably also bump the minimum version in modules/subnets to match.

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@Lynty
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by britthouser** [comment]
FWIW - I have pulled this change down locally and tested it.  It successfully added external IPv6 CIDRs to my existing subnets previously created using this module.  I then destroyed and created dualstack networks from the start using this patch.

If there's anything I can do to move this patch along and get it into a release so I can pull it in, I'd be happy to help.

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @Lynty - Can you please also add the new inputs to the module documentation: https://github.com/terraform-google-modules/terraform-google-network/blob/master/modules/subnets/README.md

**Comment by bharathkkb** [comment]
@apeabody Since custom logic seems minimal, we can add int test in a follow up if we need additional coverage.

**Review by apeabody — APPROVED** [review]

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @Lynty!

**Pull request merged**